DFE-Digital/sts-plan-technology-for-your-school
# Feat: Automated process for fetching gias data

## Overview

Addresses ticket [#236144](https://dfe-ssp.visualstudio.com/s190-schools-technology-services/_workitems/edit/236144)

Python project to fetch GIAS data and store it in our SQL database with a workflow to run this daily

## Changes

### New python project for Updating GIAS data
- Created a new python project using playwright and pyodbc for the process of fetching and updating GIAS data
  - The website dynamically generates the download file and redirects you to a different page for it, so just using requests wasn't working out and I think a small playwright script is the easiest way to handle this
  - I tried a few options for the database update and ultimately decided two temp tables for the csv data and letting a stored procedure do the update was probably the best option to minimise the updates to the fixed tables and leave it largely to the temp ones

### Pipeline
- Created a workflow to run this on a schedule
- As with my other PR, the issue with this at the moment is that it requires permission to run on prod - My suggestion for working around this is that we make another environment (or 2) for automated jobs in staging/production and give it the necessary secrets for the automated jobs only, and have these processes not require permission
- Uses webkit over chromium because for some reason chromium hangs infinitely on the download step and I spent ages trying to debug it but it runs locally just fine, webkit it is

### Update web app
- Still **TODO**
- I didn't get too far into this, but we already get the UID in the claims, so its a case of updating the `establishmentDto` and retrieving the right data from the establishmentGroup table and presenting this on the self-assessment table
- [I have a branch where I've done this](https://github.com/DFE-Digital/sts-plan-technology-for-your-school/tree/poc/236144/display-gias-name-on-self-assessment), but it was very quick and not done properly so I've reverted it from this PR 

## How to review the PR
I've already run the SQL script on dev - I'm pretty sure the SQL "risk" being flagged is a bug in sqlcheck because the foreign key is not recursive, but if you spot something I haven't please say

- Setup the python project locally, ensuring the README provides enough information to successfully get it running
- Run the project and see that the data is downloaded, extracted and the database updated
- Check that the pipeline which is running on pull_request at the moment is functioning correctly

## Release requirements
- Add the python connection string to the keyvault for test/staging/prod

## Checklist
